config/config.go

```diff
@@ -0,0 +1,137 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	c "nav/constants"
+	"os"
+)
+
+type Config struct {
+	ConfValues ConfValues
+	// add DB connection instance here
+}
+
+type ConfValues struct {
+	Symbol         string    `mapstructure:"symbol"`
+	Type           string    `mapstructure:"output_type"`
+	ExcludedBefore []string  `mapstructure:"excluded_before"`
+	ExcludedAfter  []string  `mapstructure:"excluded_after"`
+	TargetSubsys   []string  `mapstructure:"target_subsys"`
+	Database       Database  `mapstructure:"database"`
+	MaxDepth       int       `mapstructure:"max_depth"`
+	Mode           c.OutMode `mapstructure:"mode"`
+}
+
+type Database struct {
+	Name     string `mapstructure:"name"`
+	URL      string `mapstructure:"url"`
+	User     string `mapstructure:"user"`
+	Password string `mapstructure:"password"`
+	Port     int    `mapstructure:"port"`
+	Instance int    `mapstructure:"instance"`
+}
+
+// New creates a new Config instance and returns a pointer to it.
+func New() (*Config, error) {
+	config := Config{}
+	confValues, err := initConfig()
+	config.ConfValues = confValues
+	if err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+// initConfig gathers the configuration values from command line and config file (if provided) and returns
+// them in a ConfValues struct.
```

Comment:
Do command line arguments override those from the config file?

Reply:
Yes, that's the desired behavior AFAIK